contrib/XCodeCloudPipelines.md

```diff
@@ -0,0 +1,16 @@
+## Creating a new pipeline in XCode Cloud manually.
+
+Follow these steps to add a new pipeline to XCode Cloud.
+
+1. Add the new pipeline to the [PR-CI Matrix](./scripts/pr-ci-matrix.rb)! list of available targets.
+2. Add one of the following options (`all`, `latest_only` or `oldest_and_latest`) to the added target, this will dependable
+   of which version do you want to run your new workflow. The list of XCode versions available are in the same file 
+   in `XCODE_VERSIONS`.
+3. Run the following command from the project root `ruby scripts/pr-ci-matrix.rb` to generate the data on the
+   [CI_POST_CLONE](./ci_scripts/ci_post_clone.sh) file.
+4. Create the new workflow on the XCode Cloud App Store Connect dashboard (https://appstoreconnect.apple.com/teams/69a6de86-7f37-47e3-e053-5b8c7c11a4d1/frameworks/E9D174FA-8898-4C93-94F3-EDC79DF32471/workflows) or
```

Comment:
Creating the workflows really needs to be automated using the Xcode cloud API. With how many test targets we have this is a very large number of manual steps which would be very time consuming, tedious, and error-prone to do every time there's a new Xcode version.

Reply:
A general idea of how this would work:

1. [Get a list of the workflows](https://developer.apple.com/documentation/appstoreconnectapi/list_all_workflows_for_an_xcode_cloud_product) and filter them to ones with `pullRequestStartCondition` set.
2. [Delete each of them](https://developer.apple.com/documentation/appstoreconnectapi/delete_a_workflow)
3. Do something similar to the current matrix logic, but instead of emitting yaml [create a workflow for each desired combination](https://developer.apple.com/documentation/appstoreconnectapi/create_a_workflow)

Figuring out authentication and how to manage app store connect tokens is probably going to be way more obnoxious than it needs to be, but the actual API looks pretty simple.

Reply:
Added scripts for adding/modifying/deleting workflows. This can only be used with some environment values which needs to be set on each local, for it to work and having the p8 file which will be used to connect to the API